core-errors/src/simple.rs

```diff
@@ -16,180 +16,369 @@
 
 //! Simple errors being used for status codes
 
+#[cfg(feature = "codec")]
 use enum_iterator::Sequence;
 
-/// Type that can be encoded and decoded into status code
-pub trait SimpleCodec: sealed::Sealed + Sized {
-    /// Convert type into status code
-    fn into_status_code(self) -> i32;
+#[cfg(feature = "codec")]
+use scale_info::{
+    scale::{self, Decode, Encode},
+    TypeInfo,
+};
 
-    /// Convert status code into self
-    fn from_status_code(status_code: i32) -> Option<Self>;
+#[repr(u8)]
+#[derive(
+    Clone,
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Default,
+    derive_more::Display,
+    derive_more::From,
+)]
+#[cfg_attr(feature = "codec", derive(Encode, Decode, TypeInfo, Sequence), codec(crate = scale), allow(clippy::unnecessary_cast))]
```

Comment:
Isn't this clippy warning fixed in codec v3.6.1?

Reply:
Yes, it was (https://github.com/paritytech/parity-scale-codec/issues/456), but seems that we are still at earlier version within the master branch
https://github.com/gear-tech/gear/blob/dd69aff0fec5e257f5f819c6180ee0f700cd0726/Cargo.lock#L7844-L7857